texar/torch/utils/gin.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
```

Comment:
It feels weird to have this file here under `utils`, if it's written exclusively for T5 pre-trained models. Why not put it inside `pretrained/t5_utils.py`?

Reply:
Right now, only `T5` uses `gin` config files. But in the future if more pre-trained models start using this type of configuration, we might need a utility to deal with reading `gin` config files. Hence, placed it inside `utils` folder. 

Open to moving it into `pretrained/t5_utls.py` for now and then once we need more functions for reading `gin` files, can move it into `utils` function. This feels future proof though.

Reply:
But the thing is, the function here is neither really parsing `gin` nor general enough to be applied to other potential configuration files. In its current form, I would say it belongs to `t5_utils`.

Reply:
👍